src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -1555,239 +1755,47 @@ private void EndTagBlock(Stack<Tuple<SyntaxToken, SourceLocation>> tags, bool co
 
             if (!complete)
             {
-                AddMarkerTokenIfNecessary();
-            }
-            Output(SpanKindInternal.Markup);
-        }
-
-        internal static bool IsValidAttributeNameToken(SyntaxToken token)
-        {
-            if (token == null)
-            {
-                return false;
+                AcceptMarkerTokenIfNecessary();
             }
 
-            // These restrictions cover most of the spec defined: http://www.w3.org/TR/html5/syntax.html#attributes-0
-            // However, it's not all of it. For instance we don't special case control characters or allow OpenAngle.
-            // It also doesn't try to exclude Razor specific features such as the @ transition. This is based on the
-            // expectation that the parser handles such scenarios prior to falling through to name resolution.
-            var tokenType = token.Kind;
-            return tokenType != SyntaxKind.Whitespace &&
-                tokenType != SyntaxKind.NewLine &&
-                tokenType != SyntaxKind.CloseAngle &&
-                tokenType != SyntaxKind.OpenAngle &&
-                tokenType != SyntaxKind.ForwardSlash &&
-                tokenType != SyntaxKind.DoubleQuote &&
-                tokenType != SyntaxKind.SingleQuote &&
-                tokenType != SyntaxKind.Equals &&
-                tokenType != SyntaxKind.Unknown;
+            builder.Add(OutputAsMarkupLiteral());
         }
 
-        public void ParseDocument()
+        public MarkupBlockSyntax ParseRazorBlock(Tuple<string, string> nestingSequences, bool caseSensitive)
```

Comment:
Curious, why do we want this case sensitive?

Reply:
It doesn't matter in our case. Because the only nesting sequence we compare are `{` and `}`. This is just written in a generic way so that it can be used for other razor blocks where we can specify the nesting sequences.

Reply:
Feel free to nuke it if you want

Reply:
Will nuke later